src/js/ui/pokemon.js

```diff
@@ -0,0 +1,160 @@
+        Abilities: objectPropsToString(abilities, "ability") };
+
+    const $about = document.getElementById('basic-info');
+
+    Object.entries(items).forEach(item => $about.appendChild(createItem(item[0], item[1])));
+}
+
+function createSection() {
+    const $section = document.getElementById('pokemon-info-template').content.cloneNode(true);
+
+    return $section;
+}
+
+function createItem(name, value) {
+    const $itemTemplate = document.getElementById('about-item-template').content.cloneNode(true);
+
+    $itemTemplate.querySelector('h3').textContent = capitalizeFirstLetter(name);
+    $itemTemplate.querySelector('p').textContent = value;
+
+    return $itemTemplate;
+}
+
+function makeHero(data) {
+    const {name, id, sprites, types} = data;
+    const $name = document.getElementById('pokemon-name');
+    const $id = document.getElementById('pokemon-id');
+    const $img = document.getElementById('pokemon-image');
+    const $tags = document.getElementById('tags');
+
+    $name.textContent = capitalizeFirstLetter(name);
+    $id.textContent = "#" + parseToThreeDigits(id);
+    $img.src = sprites.other.dream_world.front_default;
+    appendTags($tags, createTags(getTagsNames(types)));
+
+}
+
+function createTags(names) {
+    const $tagTemplate = document.getElementById('tag-template').content;
+
+    return Array.from(names, (name) => {
+        const $newTag = $tagTemplate.querySelector('li').cloneNode(true);
+        $newTag.textContent = capitalizeFirstLetter(name);
+
+        return $newTag
+    })
+}
+
+function getTagsNames(data) {
+    const tags = Array.from(data, tag => tag.type.name );
+
+    return tags;
+}
+
+function makeStats(data) {
+    const $statsList = document.getElementById('stats-list');
+    const $statItem = document.getElementById('stats-item-template');
+    const $statTotalItem = document.getElementById('stats-total-template');
+
+    data.forEach(item => $statsList.appendChild(createStat($statItem, item.stat.name, item.base_stat)));
+
+    $statsList.appendChild(createStat($statTotalItem, "Total", data.map(item => item.base_stat).reduce((a, b) => a + b)))
+}
+
+function createStat(template, name, value) {
+    const $stat = template.content.cloneNode(true);
+
+    $stat.querySelector('h3').textContent = capitalizeFirstLetter(parseFromSnake(name));
+    $stat.querySelector('p').textContent = value;
+    $stat.querySelector('progress').value = value;
+
+    return $stat;
+
+}
+
+function getGenderProbability(genderRate) {
+    if (genderRate === -1) {
+      return [0, 0];
+    } else if (genderRate === 0) {
+      return [0, 0];
+    } else if (genderRate === 1) {
+      return [100, 0];
+    } else if (genderRate === 2) {
+      return [50, 50];
+    } else if (genderRate >= 3 && genderRate <= 7) {
+      const femaleChance = 100 / (genderRate * 2);
+      return [100 - femaleChance, femaleChance];
+    } else if (genderRate === 8) {
+      return [0, 100];
+    }
```

Comment:
Is there a way to improve this condition? What are those magic numbers? Can we save them in a variable?

Reply:
I think now it's more understandable. At that time, i was struggling with this function because the results never matched other guides values. Even today i really dont know if they are correct. The rates are eights of a chance to being female.